lua/pug/modules/lagdetection.lua

```diff
@@ -52,13 +52,27 @@ assert( type( lag.fPanic ) == "function", "Invalid PanicMethod variable" )
 	Compare new rates against the average. (Test ZScore vs Subtraction.)
 ]]
 
+
 local function getMean()
+    --local count = 0
+    --for _, val in pairs(sample.data) do
+    --    count = count + val
+    --end
+	--return count / table.Count(sample.data)
+
 	local x = tableReduce( function(a, b) return a + b end, sample.data )
 	return ( x / #sample.data )
 end
 
+local maxRate = 1 / engine.TickInterval()
+local sampleTime = 1.5 -- time in seconds to average
+local sampleSize = sampleTime * maxRate
+
 local function addSample( rate )
-	sample.index = ( sample.index % 120 ) + 1
+    rate = math.Clamp( rate, 0, maxRate )
+
+	sample.index = math.ceil( sample.index % sampleSize ) + 1
```

Comment:
I have literally no reason why, but sometimes the provided rate was _really_ high. Like, in the hundreds. It doesn't make any sense to me, but adding a clamp and a `ceil` here made sure that the provided rate never went above the server tickrate.

To me, it doesn't make sense for it to ever be higher than the server's tickrate, and even if it is, do we really care? We only care if the number goes down.

Reply:
You put the `math.ceil` in the wrong place. That's just for the index as in "where in the table should we put this value".
I have however implemented your suggestion to clamp the rate, because I guess we don't actually care if it goes up, and if it goes up as much as you're implying it might just be Lua reporting incorrect numbers.

I will make a branch where I use `engine.TickInterval` to get the interval instead of attempting to calculate it through Lua, to see if it's any better.